aten/src/ATen/native/UpSample.h

```diff
@@ -60,10 +60,15 @@ static inline void upsample_1d_shape_check(
       ")");
 
   if (input.defined()) {
+    // Allow for empty batch size but not other dimensions
+    bool valid_empty = false;
+    valid_empty = input.size(0) == 0 && input.size(1) != 0 && input.size(2) != 0;
+    
     TORCH_CHECK(
-        input.numel() != 0 && input.dim() == 3,
-        "Non-empty 3D data tensor expected but got a tensor with sizes ",
-        input.sizes());
+                (input.numel() != 0 || valid_empty) && input.dim() == 3,
```

Comment:
This seems like an awfully circuitous way to write the condition. Since dimensionality is known, just assert that `input.size(1) != 0 && input.size(2) != 0` and call it a day.